### PR TITLE
Update PDR processing to respect NODE_NAME parameter

### PR DESCRIPTION
**Pull request by yjpa7145**
**Summary:** Update PDR processing to respect NODE_NAME parameter

Addresses [CUMULUS-2141: PDR processing not respecting NODE_NAME parameter](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2141)

## Changes

* The `parse-pdr` task has been updated to respect the `NODE_NAME` property in a PDR's `FILE_GROUP`. If a `NODE_NAME` is present, the task will query the Cumulus API for a provider with that host. If a provider is found, the output granule from the task will contain a `provider` property containing that provider. If `NODE_NAME` is set but a provider with that host cannot be found in the API, or if multiple providers are found with that same host, the task will fail.
* The `@cumulus/api-client.providers.getProviders` function now takes a `queryStringParameters` parameter which can be used to filter the providers which are returned
* Bump `ava` version
* `version` is now a required property of granule input into the `queue-granules` task. (It was required by the code but not in the schema.)
* `provider` is now an optional property of granule input into the `queue-granules` task. If present, that provider will be used when enqueuing the granule. If not present, the task's `config.provider` will be used instead.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

**Review by Jkovarik — CHANGES_REQUESTED** [review]
This looks good - I've added a few comments for discussion and there's a couple of style/nit considerations.

File: example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+      testDataFolder = createTestDataPath(testId);
+
+      pdrFilename = `${testSuffix.slice(1)}_${origPdrFilename}`;
+
+      provider = { id: `s3_provider${testSuffix}` };
+
+      nodeName = `${config.stackName}-pdr-node-name-provider`;
+      nodeNameProviderId = `provider-${nodeName}`;
+
+      const createProviderResponse = await providersApi.createProvider({
```
[the remark this answers is not in the document]
Comment (yjpa7145) [reply example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js:117]: 440c1159e

File: packages/api-client/src/providers.ts [hunk 2]
```diff
@@ -97,16 +97,18 @@ export const getProvider = (params: {
  */
```
[the remark this answers is not in the document]
Comment (yjpa7145) [reply packages/api-client/src/providers.ts:101]: d571987ee

File: tasks/queue-granules/index.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ async function queueGranules(event) {
         granule,
         queueUrl: event.config.queueUrl,
         granuleIngestWorkflow: event.config.granuleIngestWorkflow,
-        provider: event.config.provider,
+        provider: granule.provider || event.config.provider,
```
[the remark this answers is not in the document]
Comment (yjpa7145) [reply tasks/queue-granules/index.js:35]: Probably, but that feels like a separate ticket.

File: tasks/parse-pdr/tests/test-parse-pdr.js [hunk 4]
```diff
@@ -85,16 +102,17 @@ test.serial('parse-pdr properly parses a simple PDR file', async (t) => {
 
   const granule = result.granules[0];
   t.is(granule.dataType, 'MOD09GQ');
+  t.is(granule.provider, undefined);
 
-  const hdfFile = result.granules[0].files.find((file) => file.name === 'MOD09GQ.A2017224.h09v02.006.2017227165020.hdf');
+  const hdfFile = granule.files.find((file) => file.name === 'MOD09GQ.A2017224.h09v02.006.2017227165020.hdf');
```
[the remark this answers is not in the document]
Comment (yjpa7145) [reply tasks/parse-pdr/tests/test-parse-pdr.js:107]: I really don't have strong feelings either way on this one.

[the remark this answers is not in the document]
Comment (Jkovarik) [reply tasks/parse-pdr/tests/test-parse-pdr.js:107]: That sounds like not near-immediate feelings of awesomeness.   👍

File: example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+          expect(lambdaOutput.payload.pdrs[0].name).toEqual(pdrFilename);
+        }
+      });
+    });
+
+    describe('the QueuePdrs Lambda', () => {
+      if (beforeAllFailed) fail('beforeAll() failed');
+      else {
+        it('has expected output', () => {
+          expect(queuePdrsOutput.payload.pdrs_queued).toEqual(1);
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js:246]: Low-effort follow-up to this: https://github.com/nasa/cumulus/pull/1875.   Will write a quick maintenance ticket to address the overall issue.

**Review by Jkovarik — APPROVED** [review]
Thanks, that addresses my concerns.   One thing outstanding is getting the updated variables into bamboo/set-bamboo-env-variables.   Approving assuming that's sorted and CI works.

**Pull request merged**